ProcessLib/LocalAssemblerTraits.h

```diff
@@ -0,0 +1,223 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_LOCALASSEMBLERTRAITS_H
+#define PROCESSLIB_LOCALASSEMBLERTRAITS_H
+
+namespace ProcessLib
+{
+
+/*! This class makes it possible to handle Eigen's block expressions transparently,
+ * both for fixed-size and dynamically allocated Eigen matrices.
+ *
+ * This class takes advantage of the instantiations given below its definition.
+ */
+template<typename ShpPol, unsigned NIntPts, unsigned NodalDOF, unsigned Dim>
+struct LocalAssemblerTraitsFixed
+{
+private:
+    template<int N, int M>
+    using Matrix = typename ShpPol::template MatrixType<N, M>;
+    template<int N>
+    using Vector = typename ShpPol::template VectorType<N>;
+
+public:
+    using ShapeMatrices = typename ShpPol::ShapeMatrices;
+
+    //! Square matrix of the given space dimensionality
+    using MatrixDimDim = Matrix<Dim, Dim>;
+
+    // TODO That only works if all process variables are single component
+    //      and use the same shape fucntions.
+    //! Local matrix for the given number of d.o.f.\ per node and number of
+    //! integration points
+    using LocalMatrix = Matrix<NIntPts*NodalDOF, NIntPts*NodalDOF>;
+    //! Local vector for the given number of d.o.f.\ per node and number of
+    //! integration points
+    using LocalVector = Vector<NIntPts*NodalDOF>;
+
+    //! Local vector for one component of one process variable.
+    //! The size is the number of nodes in the element.
+    using Vector1Comp = typename ShapeMatrices::ShapeType;
+
+    //! Laplace matrix for the given space dimensionality and number of d.o.f
+    //! per node.
+    using LaplaceMatrix = Matrix<Dim*NodalDOF, Dim*NodalDOF>;
+
+
+    //! Get a block \c Dim x \c Dim whose upper left corner is at \c top and \c left.
+    template<typename Mat>
+    static
+    typename std::enable_if<NodalDOF != 0,
```

Comment:
> Only for my understanding: If the template parameter NodelDOF is not equal zero, i.e. the size is known at compile time, blockDimDim returns a fixed size Eigen matrix? Otherwise the implementation below is activated?

Yes, if `NodalDOF != 0`, an Eigen block of fixed size is returned.
Admittedly, this is a bit hackish. I will improve the somewhat cryptc hint in the Doxygen comment of the struct

> This class takes advantage of the instantiations given below its definition.

And I will move the struct to a `detail` namespace, since it should not be used directly, but only via the `LocalAssemblerTraits` typedef.